csrc/scheduler/resize.cpp

```diff
@@ -223,6 +223,21 @@ std::unique_ptr<HeuristicParams> ResizeScheduler::computeHeuristics(
     params->largest_input = -1;
   }
 
+  // Vectorization based on the largest input if there's any input
+  // tv. This is because the current heuristics are designed to
+  // optimize the read perfornance. The largest output is used if
+  // there's no input.
+  auto ref_tv_for_vectorization =
+      largest_input != nullptr ? largest_input : largest_output;
```

Comment:
Does this make a difference, whether we start from the input or output for vectorization analysis? My naive understanding is that the analysis would propagate it from the ref to all IO TVs.

Reply:
No, I don't think it would make any difference. I think I originally did this because there was a segment with a concat and a transpose. I think I wanted to vectorize the innermost axis of the input, which corresponds to an non-innermost axis of the output of the segment.

This actually is no longer relevant as the concat and transpose pattern is now handled as a pointwise segment after #3687. Also I added a check to reject transpose in `ResizeScheduler::canScheduleCompileTime` as it isn't tuned for transpose. I'll update the code.

Reply:
@jjsjann123 I found that it actually needs to use the largest input because we cancel reshapes of the tensors before vectorization. This is to some extent the current vectorization is only done using the innermost dimension. The innermost dimension of the reference tensor may be first "unsplit" to cancel a reshape. 

This should be fine for RoPE, but we should definitely need to revisit to make the scheduler more generic.